### ISSUE TITLE
[$1000] Workspace - Margin of error is slightly shifted to the left compared to other items

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#22242]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/21662
## Action Performed:
1. Open Desktop app
2. Create a Workspace X as user A
3. Invite user B to the Workspace
4. Login as same user A on Web
5. Copy URL of the created Workspace X on Web
6. Login as user B in incognito
7. As user B create a public room for Workspace X
8. As user A on Desktop app delete the Workspace X
9. As user A on Web Open link from step 5
10. As user A on Web accept redirection to desktop app

## Expected Result:
The margin of error is consistent with the workspace item margin

## Actual Result:
The margin of the error is slightly shifted to the left (Red circle is not aligned with other items)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.36.4

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![Bug6116580_photo_2023-07-05_03-08-45](https://github.com/Expensify/App/assets/78819774/fe77b64a-dd38-4340-be47-88bd546693bf)

https://github.com/Expensify/App/assets/78819774/9681a42a-96df-44b0-b9ae-ad742023ded8

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014ed9881792dde5c5</li>
        <li>Upwork Job ID: 1676594232511758336</li>
        <li>Last Price Increase: 2023-07-05</li>
    </ul>
</details>

**Comment by jeet-dhandha** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Auth DeletePolicy error's Dot indicator is not aligned properly with other icons.

### What is the root cause of that problem?
- Margin is not set properly for the Dot indicator.

### What changes do you think we should make in order to solve the problem?
- We can set `marginLeft: 10` in `errorRowStyles` of `OfflineWithFeedback` in `WorkspaceInitialPage.js`.
- I tried using default values like ml2, ml3. But `10` value was perfect for this scenario.

<details>
<summary> Screenshot after changes </summary>

<img width="289" alt="Screenshot 2023-07-05 at 10 28 41 AM" src="https://github.com/Expensify/App/assets/78416198/a7c382f7-ddf9-49c2-a7c6-6598efd7d0ef">

</details>


### What alternative solutions did you explore? (Optional)
- N/A

**Comment by Nikhil-Vats** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace - Dot error icon is not left aligned with other items.

### What is the root cause of that problem?
`WorkspaceInitialPage` uses `MenuItem` which uses `styles.popoverMenuIcon` for styling icons of menu items.
https://github.com/Expensify/App/blob/11b8908b8ba3282d06543e2808a7c722690521fa/src/components/MenuItem.js#L152

`styles.popoverMenuIcon` has a width of `40px` and alignment properties for centering the icon horizontally and vertically.
https://github.com/Expensify/App/blob/11b8908b8ba3282d06543e2808a7c722690521fa/src/styles/styles.js#L1309-L1313

The error message on `WorkspaceInitialPage` is shown by passing `errors={policy.errors}` to `OfflineWithFeedback` which renders `DotIndicatorMessage`.
https://github.com/Expensify/App/blob/11b8908b8ba3282d06543e2808a7c722690521fa/src/components/DotIndicatorMessage.js#L57-L62

`DotIndicatorMessage` uses only `styles.errorDot` for styling container of icon due to which there is a mismatch in the styles used for menu item icons and error dot icon. 
https://github.com/Expensify/App/blob/11b8908b8ba3282d06543e2808a7c722690521fa/src/styles/styles.js#L2846-L2848

### What changes do you think we should make in order to solve the problem?
We can't simply add `styles.popoverMenuIcon` to container of icon in `DotIndicatorMessage` as it will break styling on other pages.

We should create a new prop named `errorIconContainerStyles` in `OfflineWithFeedback` and `DotIndicatorMessage`.
From `WorkspaceInitialPage` we can pass a new prop to `OfflineWithFeedback` -
```javascript
errorIconContainerStyles={[styles.popoverMenuIcon]}
```

Then in `OfflineWithFeedback` we simply pass on the prop to `DotIndicatorMessage`.

In `DotIndicatorMessage` we can add this prop to the container of icon. We can change line 57 to -
```javascript
<View style={[styles.offlineFeedback.errorDot, ...props.errorIconContainerStyles]}>
```

This will fix the issue while also not causing any regressions on other pages.

### What alternative solutions did you explore? (Optional)
NA

**Comment by Nikhil-Vats** [comment]
Sure, I will come up with a list today and update the proposal.

**Comment by Nikhil-Vats** [comment]
@thesahindia Here is the list of components that use `OfflineWithFeedback` to show errors -

<details><summary>1. IOUPreview</summary>
<p>

<img width="1065" alt="image" src="https://github.com/Expensify/App/assets/31413407/8df581aa-4027-4d7c-ab5c-eb045941a8e9">


</p>
</details> 

<details><summary>2. ReportActionItem</summary>
<p>

<img width="1066" alt="Screenshot 2023-07-07 at 2 13 37 PM" src="https://github.com/Expensify/App/assets/31413407/78acf0fa-82fa-4486-adf7-bb27fcec6009">


</p>
</details> 

<details><summary>3. ReportActionItemCreated</summary>
<p>

<img width="1065" alt="image" src="https://github.com/Expensify/App/assets/31413407/417e91e5-34d7-4ad3-bf00-47fba974b6f4">

</p>
</details> 

<details><summary>4. ReportActionItemParentAction</summary>
<p>

<img width="1065" alt="image" src="https://github.com/Expensify/App/assets/31413407/6eb3f888-1cb1-4971-85e3-9b1aba3476e0">

</p>
</details> 

<details><summary>5. ContinueBankAccountSetup </summary>
<p>

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/fc821070-4359-4ac3-93fd-95be0be61269">

</p>
</details> 

<details><summary>6. EnableStep </summary>
<p>

<img width="374" alt="image" src="https://github.com/Expensify/App/assets/31413407/a227a7e0-57dd-4c34-a010-8be7b205c2b4">

</p>
</details> 

<details><summary>7. PaymentMethodList </summary>
<p>

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/11ea4c16-a0e3-4fe6-a406-ec20e45bb154">

</p>
</details> 

<details><summary>8. BasePaymentsPage </summary>
<p>

<img width="376" alt="image" src="https://github.com/Expensify/App/assets/31413407/b9e9ed47-8e59-4de5-8e9b-323048a3e7b6">

</p>
</details> 

<details><summary>9. ContactMethodDetailsPage </summary>
<p>

<img width="376" alt="image" src="https://github.com/Expensify/App/assets/31413407/3cedfa32-886f-4ed9-bba2-dc2c09a5b658">

<img width="377" alt="image" src="https://github.com/Expensify/App/assets/31413407/3c91e41e-f39a-4a64-8b63-3a8e1c116dad">

</p>
</details> 

<details><summary>10. BaseValidateCodeForm </summary>
<p>

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/fb2acaba-03f2-4968-8de4-19ae9b9dab82">

</p>
</details> 

<details><summary>11. ReportSettingsPage </summary>
<p>

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/18679860-187a-460d-8c76-697454a7c210">

</p>
</details> 

<details><summary>12. WorkspaceInitialPage </summary>
<p>

<img width="374" alt="image" src="https://github.com/Expensify/App/assets/31413407/f1df6838-6444-4b3e-b33b-7677a6e75315">

</p>
</details> 

<details><summary>13. WorkspaceMembersPage </summary>
<p>

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/f91d0f9c-0708-4cbc-95a0-547ef3296d41">

</p>
</details> 

<details><summary>14. WorkspacesListPage </summary>
<p>

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/51f21db4-0867-4c94-9a74-fabfe046a4a5">

</p>
</details> 


<details><summary>15. WorkspaceRateAndUnitPage </summary>
<p>

<img width="378" alt="image" src="https://github.com/Expensify/App/assets/31413407/6349dd32-09cf-4f11-b705-5d28903a8731">

</p>
</details> 

<details><summary>16. AvatarWithImagePicker - Definitely needs fixing</summary>
<p>

We were not showing errors for this component before but in [this PR](https://github.com/Expensify/App/pull/20288), AvatarWithImagePicker was changed to use OfflineWithFeedback and errors prop was passed (though it had nothing to do with PR) so that's why this looks completely messed up because it was not intentionally added -
<img width="375" alt="image" src="https://github.com/Expensify/App/assets/31413407/f9560073-7a92-41d8-bf55-51cd22dfe457">

</p>
</details>

**Comment by Nikhil-Vats** [comment]
The following pages have alignment issues in my opinion - 
5, 6 - left align the red dot with icons in buttons
8 - both errors need alignment
12 (this issue) - left align red dot with icons
13, 15 - align close icon
16 - remove error prop as we were not showing it before and it was added in [this PR](https://github.com/Expensify/App/pull/20288) unintentionally.

**Comment by puneetlath** [comment]
@madmax330 thoughts on the proposal?

**Comment by madmax330** [comment]
Cc @shawnborton can you confirm that this looks good before we move forward?

**Comment by shawnborton** [comment]
I actually don't think this is a bug. We used to try to get cheeky with making sure the red dot lined up with the content above it, but we ran into way too many different cases where sometimes the item above was further right than others. So in the spirit of consistency, we just made it so we have one default RBR error message style, which is what you see. Persnoally I don't think this looks broken, and I think we should just close this out.
